pkg/api/pod/util.go

```diff
@@ -1260,3 +1261,56 @@ func MarkPodProposedForResize(oldPod, newPod *api.Pod) {
 		}
 	}
 }
+
+// KEP: https://kep.k8s.io/4639
+func dropImageVolumes(podSpec, oldPodSpec *api.PodSpec) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) || imageVolumesInUse(oldPodSpec) {
```

Comment:
shouldn't this be !imageVolumesInUse ?

Reply:
Hm, no the logic seems to be correct looking at the other functions and also test cases. Am I missing something?

Reply:
if featureGate: don't drop
if !imageVolumesInUse: nothing to drop
if either are not set, then don' t we need to drop? as in: if image volumes are being used but the feature is disabled, wouldn't we need to clear them?

Reply:
Hm, I was looking how others implemented it and it’s the same logic for `dropPodLifecycleSleepAction`, right?

https://github.com/kubernetes/kubernetes/blob/14592ef39d6c08fb5af48df3edadb4556dee2e9a/pkg/api/pod/util.go#L720

Reply:
that one also looks wrong haha for instance here's userns https://github.com/kubernetes/kubernetes/blob/14592ef39d6c08fb5af48df3edadb4556dee2e9a/pkg/api/pod/util.go#L650

Reply:
the majority of them check !inUse AFAICT

Reply:
We drop if `!enabled && !inuse`, means we don’t drop if `!(!enabled && !inuse)` which is the same as `enabled || inuse`, right? Am I wrong?

Reply:
okay you're right, sorry for the noise. I think I still don't understand why this is the way this code works, but you're definitely following precedent and it must be there for a reason I'm missing :upside_down_face:

Reply:
It’s tricky but I was aligning with the existing logic and test cases. So I assume that’s correct. 👍